src/script/lua_api/l_base.cpp

```diff
@@ -125,8 +125,11 @@ int ModApiBase::l_deprecated_function(lua_State *L, const char *good, const char
 	lua_Debug ar;
 
 	// Get caller name with line and script backtrace
-	FATAL_ERROR_IF(!lua_getstack(L, 1, &ar), "lua_getstack() failed");
-	FATAL_ERROR_IF(!lua_getinfo(L, "Sl", &ar), "lua_getinfo() failed");
+	if (!lua_getstack(L, 1, &ar) || !lua_getinfo(L, "Sl", &ar)) {
+		errorstream << __func__ << ": lua_getstack or lua_getinfo for call '" <<
+			bad << "' failed. " << script_get_backtrace(L) << std::endl;
```

Comment:
well what about the deprecated warning?

Reply:
It's kinda pointless to have that if we cannot point the user to a proper code location...

Reply:
it is not. the proper solution is to tell the user what we most probably know: that this is a deprecated *tail call* in a *coroutine*. there will not be a lot of these.

i think the proper solution is to just keep the warning, and instead of a precise location, give the aforementioned hint.

Reply:
btw can't we reuse `script_log_add_source` here? that just says `(at ?:?)` in this case. maybe it should give this hint too.

Reply:
Yes, I hope that suffices to narrow it down. Updated the PR accordingly.